### PR TITLE
Use poetry to merge requirements files

### PR DESCRIPTION
**Pull request by BurnySc2**
- Add github action tests
- Use poetry to merge client-requirements and bot-requirements
- Remove conflicting libs
- Sc2 linux base image now uses dotnet 6.0 instead of 2.2

~~This will require https://github.com/aiarena/aiarena-client/pull/56 to be merged before tests run through~~

~~Can https://github.com/aiarena/aiarena-docker/blob/master/.gitlab-ci.yml be removed?~~

Needs:
Merge of https://github.com/aiarena/rust-arenaclient/pull/37

Removal of the following libraries:
- async-timeout
- chardet
- idna

image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,44 +1,69 @@
...
+RUN echo $SC2_VERSION
+RUN echo $VERSION_NUMBER
+RUN echo $USE_SQUASHED
 
-# Calls for a random number to break the cahing of the git clone
-# (https://stackoverflow.com/questions/35134713/disable-cache-for-specific-run-commands/58801213#58801213)
-ADD "https://www.random.org/cgi-bin/randbyte?nbytes=10&format=h" skipcache
+WORKDIR /root/
+
+# Prevent caching unless client master branch changed
```
Comment (lladdy) [inline Dockerfile:18]: Is this a typo?
Should it say "Prevent caching **when** client master branch changed"?

If so, very useful - been wanting a way to do this.

File: Dockerfile [hunk 2]
```diff
@@ -1,44 +1,69 @@
...
 # Add Pythonpath to env
 ENV PYTHONPATH=/root/aiarena-client/:/root/aiarena-client/arenaclient/
-ENV HOST 0.0.0.0
+ENV HOST=0.0.0.0
 
 # Install the arena client as a module
-RUN python3.9 /root/aiarena-client/setup.py install
-
-# Add Pythonpath to env
-ENV PYTHONPATH=/root/aiarena-client/:/root/aiarena-client/arenaclient/
```
Comment (lladdy) [inline Dockerfile:39]: Does it matter that we're removing this?
Just checking it's not inadvertent.
I guess if the tests work, then it's not required.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+on: [push, pull_request]
+
+# Version number of the docker images - increment when there are huge changes
+env:
+  VERSION_NUMBER: '1.0.13'
+  LATEST_PYTHON_VERSION: '3.9'
+  LATEST_SC2_VERSION: '4.10'
+
+# TODO Enable python-version '3.10' in all workflow matrices when https://github.com/aiarena/rust-arenaclient/pull/37 was merged
```
Comment (lladdy) [inline .github/workflows/ci.yml:11]: I've requested a review from Dan on this PR.

**Review by lladdy — COMMENTED** [review]
> Sc2 linux base image now uses dotnet 6.0 instead of 2.2
Just checking: Any idea if this will run bots for dotnet 5 as well?

File: Dockerfile (hunk 1 above)
Comment (BurnySc2) [reply Dockerfile:18]: Yes, this is a typo

File: Dockerfile (hunk 2 above)
Comment (BurnySc2) [reply Dockerfile:39]: The install command was renamed
image


And the env was listed twice
image

File: .github/workflows/ci.yml (hunk 3 above)
Comment (BurnySc2) [reply .github/workflows/ci.yml:11]: He should be aware but just doesn't seem to have the time at the moment

**Comment by BurnySc2** [comment]
Reply to https://github.com/aiarena/aiarena-docker/pull/91#pullrequestreview-985677355
> Sc2 linux base image now uses dotnet 6.0 instead of 2.2
> Just checking: Any idea if this will run bots for dotnet 5 as well?

Yes the 2.2 support was dropped, but the image should contain dotnet 5 and dotnet 6 runetime at the moment
It would be ideal to have (compiled) bots from other languages (not just python) in https://github.com/aiarena/aiarena-test-bots 
Then these could be added to the list of tests to run
https://github.com/aiarena/aiarena-client/blob/b8cbd4660acb65e032d7812b688fe67a13f5a4ab/testing/test_matches_aiarena_client_bots.json#L2-L15

**Review by lladdy — APPROVED** [review]

**Pull request merged**